src/extensions/renderer/canvas/drawing-edges.js

```diff
@@ -32,8 +32,18 @@ CRp.drawEdge = function( context, edge, shiftToOriginWithBb, drawLabel ){
     context.lineWidth = edgeWidth;
     context.lineCap = 'butt';
 
-    r.strokeStyle( context, lineColor[0], lineColor[1], lineColor[2], strokeOpacity );
-
+    if (Array.isArray(lineColor[0])) { //check if gradient
+      let grd = context.createLinearGradient(rs.allpts[0], rs.allpts[1], rs.allpts[rs.allpts.length-2], rs.allpts[rs.allpts.length-1]);
```

Comment:
(5) If you're not iterating, you shouldn't use `rs.allpts`.  Use `rs.source` and `rs.target`.